esmvalcore/preprocessor/_multimodel.py

```diff
@@ -429,10 +349,7 @@ def multi_model_statistics(products, span, statistics, output_products=None):
 
     for statistic in statistics:
         # Compute statistic
-        if span == 'overlap':
-            statistic_cube = _assemble_overlap_data(cubes, interval, statistic)
-        elif span == 'full':
-            statistic_cube = _assemble_full_data(cubes, statistic)
+        statistic_cube = _assemble_data(cubes, statistic, span)
         statistic_cube.data = np.ma.array(statistic_cube.data,
```

Comment:
This is not new in the pull request, but why is the array converted (again) to a masked array here? It would also be much more memory efficient to ensure that the data type is correct during the calculation already.

Reply:
I changed that in my last commit.